components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -1283,6 +1288,56 @@ void AtmosphereDriver::set_initial_conditions ()
     }
   }
 
+  // Compute IC perturbations of GLL fields (if requested)
+  using vos = std::vector<std::string>;
+  const auto perturbed_fields = ic_pl.get<vos>("perturbed_fields", {});
+  const auto num_perturb_fields = perturbed_fields.size();
+  if (num_perturb_fields > 0) {
+    m_atm_logger->info("    [EAMxx] Adding random perturbation to ICs ...");
+
+    EKAT_REQUIRE_MSG(m_field_mgrs.count("Physics GLL") > 0,
+                     "Error! Random perturbation can only be applied to fields on "
+                     "the GLL grid, but no Physics GLL FieldManager was defined.\n");
+    const auto& fm = m_field_mgrs.at("Physics GLL");
+
+    // Setup RNG. If no random seed is requested in
+    // initial_conditions params, generate one using rand().
+    const auto seed = ic_pl.get<int>("perturbation_random_seed", rand());
+    std::mt19937_64 engine(seed);
+    m_atm_logger->info("      For IC perturbation, random seed: "+std::to_string(seed));
+
+    // Get perterbation limit. Defines a range [-perturbation_limit, perturbation_limit]
+    // for which the perturbation value will be randomly generated from.
+    const auto perturbation_limit = ic_pl.get<Real>("perturbation_limits", 0.001);
+
+    // Define a level mask using reference pressure and the pressure_lower_bound parameter.
+    // This mask dictates which levels we apply a perturbation.
+    const auto gll_grid = m_grids_manager->get_grid("Physics GLL");
+    gll_grid->get_geometry_data("hyam").sync_to_host(), gll_grid->get_geometry_data("hybm").sync_to_host();
+    const auto hyam = gll_grid->get_geometry_data("hyam").get_view<const Real*, Host>();
+    const auto hybm = gll_grid->get_geometry_data("hybm").get_view<const Real*, Host>();
+    constexpr auto ps0 = physics::Constants<Real>::P0;
+    const auto pressure_lower_bound = ic_pl.get<Real>("perturbation_pressure_lower_bound", 1050.0);
+    auto pressure_mask = [&] (const int ilev) {
+      const auto pref = (hyam(ilev)*ps0 + hybm(ilev)*ps0)/100; // Reference pressure ps0 is in Pa, convert to milibar
+      return pref > pressure_lower_bound;
+    };
+
+    // Loop through fields and apply perturbation.
+    for (size_t f=0; f<perturbed_fields.size(); ++f) {
+      const auto fname = perturbed_fields[f];
+      EKAT_REQUIRE_MSG(ekat::contains(fields_inited[fm->get_grid()->name()], fname),
+                       "Error! Attempting to apply perturbation to field not in initial_conditions.\n"
+                       "  - Field: "+fname+"\n"
+                       "  - Grid:  "+fm->get_grid()->name()+"\n");
+
+      auto field = fm->get_field(fname);
+      std::uniform_real_distribution<Real> pdf(-1.0*perturbation_limit, perturbation_limit);
+      perturb(field, engine, pdf, pressure_mask);
```

Comment:
Can we specify the random seed? Doing so would allow us to BFB repeat a particular simulation if needed, which would be nice. Like, the random seed for each run could be either a user-input or randomly generated. If it is alwayas printed to the log file, we could repeat any run we want. Though I'm not sure if we'd need a different seed for each variable we're randomizing or even each col and lev within a field... that would be a hassle.